crates/extensions/c8y_http_proxy/src/actor.rs

```diff
@@ -187,31 +195,113 @@ impl C8YHttpProxyActor {
         device_id: Option<&str>,
     ) -> Result<String, C8YRestError> {
         let url_get_id = self.end_point.get_url_for_get_id(device_id);
+        let req_build_closure = |jwt_token: String, _internal_id: String| {
+            let request_internal_id = HttpRequestBuilder::get(url_get_id.clone());
+            request_internal_id.bearer_auth(jwt_token)
+        };
+
+        let token = if let Ok(token) = self.peers.jwt.await_response(()).await? {
+            self.end_point.token = token.clone();
+            token
+        } else {
+            return Err(C8YRestError::CustomError("JWT token not available".into()));
+        };
```

Comment:
This must be moved inside the execute method: this method is responsible of getting a token and refreshing it if required.

Reply:
done